integration-tests/src/test/java/com/datastax/oss/driver/mapper/NullSavingStrategyIT.java

```diff
@@ -66,14 +69,30 @@ public class NullSavingStrategyIT {
 
   @BeforeClass
   public static void setup() {
-    CqlSession session = SESSION_RULE.session();
-    session.execute(
+    // Create V3 protocol client for use in the tests
+    v3Session =
+        SessionUtils.newSession(
+            CCM_RULE,
+            SESSION_RULE.keyspace(),
```

Comment:
An interesting question (that I genuinely don't know the answer to)... is this always safe?  We're accessing a static field from a static method which presumes that the field will always be initialized before the method is called... even though both are static.  This isn't a _change_ in behaviour (the old code did it too)... just something I wondered about.

Reply:
It depends on the junit runner. In this case we're using the default which ends up being `ParentRunner`. Jumping into the debugger that uses this `classBlock()` method to determine the order of operation:

> If there are any children remaining after filtering and ignoring, construct a statement that will:
> * Apply all ClassRules on the test-class and superclasses.
> * Run all non-overridden @BeforeClass methods on the test-class and superclasses; if any throws an Exception, stop execution and pass the exception on.
> * Run all remaining tests on the test-class.
> * Run all non-overridden @AfterClass methods on the test-class and superclasses: exceptions thrown by previous steps are combined, if necessary, with exceptions from AfterClass methods into a [MultipleFailureException]
> 

https://junit.org/junit4/javadoc/4.13/org/junit/runners/ParentRunner.html#classBlock(org.junit.runner.notification.RunNotifier)

So the order will be `CcmRule#before()` > `SessionRule#before()` > `NullSavingStrategyIT#setup()`